### PR TITLE
armor 0.1.1 (new formula)

### PR DESCRIPTION
**Pull request by vishr**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

Signed-off-by: Vishal Rana vr@labstack.com

### CODE REVIEW

File: Formula/armor.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    armorpath = buildpath/"src/github.com/labstack/armor"
+    armorpath.install buildpath.children
+
+    cd armorpath do
+      system "go", "build", "-o", bin/"armor", "cmd/armor/main.go"
+    end
+  end
+
+  test do
+    system "armor", "-v"
```
Comment (MikeMcQuaid) [inline Formula/armor.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Comment (MikeMcQuaid) [inline Formula/armor.rb]: Regardless `system "#{bin}/armor"` is needed

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/armor.rb (hunk 1 above)
Comment (vishr) [inline Formula/armor.rb]: Changes done.

File: Formula/armor.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    armorpath = buildpath/"src/github.com/labstack/armor"
+    armorpath.install buildpath.children
+
+    cd armorpath do
+      system "go", "build", "-o", bin/"armor", "cmd/armor/main.go"
```
Comment (DomT4) [inline Formula/armor.rb]: Will need a:
```ruby
prefix.install_metafiles
```
line underneath this as well.

File: Formula/armor.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/armor -v")
```
Comment (dunn) [inline Formula/armor.rb]: I think Mike meant a more substantial test than checking the `-v` output.   Can you start the server, see if the basic functionality is working?

Comment (vishr) [inline Formula/armor.rb]: My knowledge with ruby is limited, do you think we can defer this and get the formula in?

Comment (dunn) [inline Formula/armor.rb]: Try looking at formulae like [`mailhog.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/mailhog.rb) and [`algernon.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/algernon.rb) and [`etcd.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/etcd.rb) for some examples, and let us know if you have questions!

Comment (vishr) [inline Formula/armor.rb]: @dunn Thanks for the examples, I followed one of them and updated the test.

File: Formula/armor.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    begin
+      pid = fork do
+        exec "#{bin}/armor"
+      end
+      sleep(1)
```
Comment (bfontaine) [inline Formula/armor.rb]: Small nit but you don’t need the parentheses here: `sleep 1`.

Comment (vishr) [inline Formula/armor.rb]: @bfontaine It is fixed now.

**Comment by vishr** [comment]
Whats the expected time for this to get merged? Please let me know if anymore changes are required at my end.

**Comment by dunn** [comment]
Merged, thank you!!

**Pull request closed without merging**